readthedocs/oauth/migrations/0014_remove_remoterepository_project.py

```diff
@@ -0,0 +1,27 @@
+# Generated by Django 2.2.22 on 2021-06-14 15:42
+
+from django.db import migrations
+
+
+def migrate_data(apps, schema_editor):
+    RemoteRepository = apps.get_model('oauth', 'RemoteRepository')
+    queryset = RemoteRepository.objects.filter(project__isnull=False).select_related('project')
+    for rr in queryset.iterator():
+        rr.project.remote_repository = rr
+        rr.save()
```

Comment:
Took me a second to follow this logic, but I *think* this looks good. I suppose I would have concerns about performance here, we will have a few thousand remote repositories to step through individually.

Reply:
I just checked this: 22k. We should be fine.

Reply:
I double-checked this logic and it didn't work.

```
rr.project.remote_repository = rr
```

That doesn't seem to keep the relation.

I also tried, 

```
rr.project.remote_repository_id = rr.pk
```

I don't understand why.

Reply:
Meh, I found that I was just saving the wrong object.

We need either:

```
rr.project.save()`
```

or

```
rr.projects.add(rr.project)
```

However, both are producing an error that I'm not sure to understand 😞 

```
database_1  | 2021-06-17 12:05:45.877 UTC [193] ERROR:  cannot ALTER TABLE "projects_project" because it has pending trigger events
database_1  | 2021-06-17 12:05:45.877 UTC [193] STATEMENT:  SET CONSTRAINTS "oauth_remotereposito_project_id_246c363d_fk_projects_" IMMEDIATE; ALTER TABLE "oauth_remoterepository_2020" DROP CONSTRAINT "oauth_remotereposito_project_id_246c363d_fk_projects_"
```